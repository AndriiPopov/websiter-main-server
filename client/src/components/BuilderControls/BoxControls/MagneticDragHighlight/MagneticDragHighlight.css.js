export const HighlightLine = {
    background: '#ff03be',
    position: 'fixed',
    zIndex: '16777269'
};

export const HighlightElement = {
    outline: '1px solid #ff03be',
    position: 'fixed',
    zIndex: '16777269',
    background: 'transparent'
};