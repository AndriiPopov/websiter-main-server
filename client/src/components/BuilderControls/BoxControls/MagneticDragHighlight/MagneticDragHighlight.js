/*
    This creates a line and highlights the element that is magnetized to on box drag
    It is used for both horizontal and vertical highlights
*/

import React from 'react';
import PropTypes from 'prop-types';

import * as classes from './MagneticDragHighlight.css.js';
import AUX from '../../../../hoc/Aux';

export const MagneticDragHighlight = props => {
    return (
        <AUX>
            <div style={{...classes.HighlightLine,
                width: props.line.width + 'px',
                height: props.line.height + 'px',
                left: props.line.left + 'px',
                top: props.line.top + 'px'}} />
            <div style={{...classes.HighlightElement,
                width: props.element.width + 'px',
                height: props.element.height + 'px',
                left: props.element.left + 'px',
                top: props.element.top + 'px'}} />
        </AUX>
    );
};

MagneticDragHighlight.propsTypes = {
    line: PropTypes.shape({
        width: PropTypes.number.isRequired,
        height: PropTypes.number.isRequired,
        left: PropTypes.number.isRequired,
        top: PropTypes.number.isRequired,
    }),
    element: PropTypes.shape({
        width: PropTypes.number.isRequired,
        height: PropTypes.number.isRequired,
        left: PropTypes.number.isRequired,
        top: PropTypes.number.isRequired,
    })
}

export default MagneticDragHighlight;