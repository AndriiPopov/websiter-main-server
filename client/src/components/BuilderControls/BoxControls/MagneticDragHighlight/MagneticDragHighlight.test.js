import React from 'react';
import {shallow, configure} from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import { MagneticDragHighlight } from './MagneticDragHighlight';

configure({adapter: new Adapter()});

describe('MagneticDragHighlight', () => {
    it('should be defined', () => {
        expect(MagneticDragHighlight).toBeDefined();
    });

    it('it should render 2 divs', () => {
        const tree = shallow(
            <MagneticDragHighlight 
                line={{
                    height: 100,
                    width: 200,
                    left: 300,
                    top: 500
                }}
                element={{
                    height: 101,
                    width: 201,
                    left: 301,
                    top: 501
                }} />
        );
        expect(toJson(tree)).toMatchSnapshot();
    });
});