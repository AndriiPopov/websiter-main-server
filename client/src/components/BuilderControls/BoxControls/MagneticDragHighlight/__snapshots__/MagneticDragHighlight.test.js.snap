// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MagneticDragHighlight it should render 2 divs 1`] = `
<aux>
  <div
    style={
      Object {
        "background": "#ff03be",
        "height": "100px",
        "left": "300px",
        "position": "fixed",
        "top": "500px",
        "width": "200px",
        "zIndex": "16777269",
      }
    }
  />
  <div
    style={
      Object {
        "background": "transparent",
        "height": "101px",
        "left": "301px",
        "outline": "1px solid #ff03be",
        "position": "fixed",
        "top": "501px",
        "width": "201px",
        "zIndex": "16777269",
      }
    }
  />
</aux>
`;
